### ISSUE TITLE
Add CompCert to the picolibc CI (somewhen)

### ISSUE THREAD

**Issue by archi** [picolibc/picolibc#112]
Hi Keith, as mentioned in issue #111, getting CompCert into your CI for regression testing is on my TODO list.
Overall, I think it will only really make sense once a stable version of CompCert is able to build your `main` branch - and since I'm currently waiting on some internal feature branches being merged, this might still take a while.

Anyway, I talked to my boss and he agrees that it's a good idea and we should give you access to our builds. Bureaucracy-wise he said we can just give you access, no NDA or other formal agreement necessary (just don't publish the build and use it only for the CI - CompCert is open source, but not free software. If SiFive wants to play around with CompCert, or use/evaluate it, an agreement would be necessary I'm afraid).

We have two packages: The first is the "**normal**" package, only containing the pre-built CompCert - that's what we give to customers. They then have to add a GCC, binutils and set up the configuration files.
Our "**internal**" package additionally contains these (as well as a build of our branch of picolibc) and is completely preconfigured. 

I think both are suited for the CI, but the internal is much easier to use (unzip and just call the compiler), so I'd recommend using that. To comply with the various opensource licenses, we can make the sources available (though I plan to upstream the picolibc changes anyway, and for riscv we use the vanilla crosstools-ng build of GCC/binutils/..., so nothing interesting there :wink:).

As mentioned above, I think actually running CompCert in the CI only really makes sense once a `main` branch CompCert can build the `main` branch picolibc. Once that's running, we could then update it on every major release (twice a year, April and October). Unlike GCC, keeping old versions doesn't make much sense, imho.

So, to summarize:
1. Which build to use, normal or internal?
2. Where to put the package so the CI can access it?
3. When to update CompCert?

If you want to play around with the compiler, drop me a mail at `${my_lastname}@absint.com`, and I will send you a recent nightly build. Though I still have some outstanding meson patches I need to submit upstream (I can just send them along, plus a snapshot of my magic perl script for building picolibc).

Let me know what you think :)

**Comment by keith-packard** [comment]
This all sounds good. Let's plan on using the internal version on the github CI infra once you're ready to have it run against the main branch. We can create a test branch that includes the updated CI bits and work out how to provide suitable access to the bits from the github environment; I believe there's a way to store secrets in the repository we could use to authenticate access from there. If you like, you can even test this in your own repository before submitting a PR and we can get the secrets transferred under separate cover. I'll let you decide when you want this to happen.

I'll see if SiFive has any interest in this; we're rather busy there with Linux stuff at present, so it may be a while :-)
